### PR TITLE
Assoc with product

### PR DESCRIPTION
**Pull request by threadhead**
Adding relations from a `LineItemPersonalization` back to:

- `Spree::ProductPersonalization`

- `Spree::OptionValueProductPersonalization`

### CODE REVIEW

File: spec/models/spree/line_item_spec.rb [hunk 1]
```diff
@@ -22,15 +22,15 @@
...
   end
 
   context "validation" do
     it "is invalid when required personalization is not set" do
       @product_personalizations[0].required = true
       line_item = @order.contents.add(@variant, @quantity, get_params([@personalization_2, @personalization_4]))
-      expect(line_item.valid?).to be_false
-      expect(line_item.errors.messages[:'personalizations.missing'].size > 0).to be_true
+      expect(line_item.valid?).to eq false
+      expect(line_item.errors.messages[:'personalizations.missing'].size > 0).to eq true
```
Comment (bukue) [inline spec/models/spree/line_item_spec.rb]: can we test for the actual size rather than greater than zero?

Comment (threadhead) [inline spec/models/spree/line_item_spec.rb]: Of course. I was just fixing deprecations here.

File: spec/models/spree/line_item_personalization_spec.rb [hunk 2]
```diff
@@ -35,25 +35,39 @@ def get_params(personalization_attributes)
...
 
         @line_item_personalization.value = "long"
-        expect(@line_item_personalization.valid?).to be_true
+        expect(@line_item_personalization.valid?).to eq true
       end
     end
+  end
+
+  it 'has relation to product personalization' do
+    @order.contents.add(@variant, @quantity, get_params([@personalization_1, @personalization_4]))
```
Comment (threadhead) [inline spec/models/spree/line_item_personalization_spec.rb]: Yup. There are a LOT of those that could be converted.

If I have the time this weekend, I may do it.

**Comment by threadhead** [comment]
You will have to run specs locally to verify that they all pass. Follow the instructions in the README.md.

**Comment by mjaniszewski** [comment]
lgtm

**Pull request merged**